### PR TITLE
Purge on Commit for Relevant Cases

### PR DESCRIPTION
**Pull request by ctsims**
Adds (as a feature flag for now) the ability to trigger a purge whenever the case xml processor detects that the casedb now has stale cases.

CommCare Field/Dev Announce:

Adds a feature flag (cc-auto-purge: yes) for triggering casedb purges when a form is completed that might result in cases needing to leave the device, for instance because their owner ID was changed.

cross-request: https://github.com/dimagi/commcare/pull/265

### CODE REVIEW

File: app/src/org/commcare/xml/AndroidTransactionParserFactory.java [hunk 1]
```diff
@@ -87,7 +88,13 @@ public void initCaseParser() {
             @Override
             public TransactionParser<Case> getParser(KXmlParser parser) {
                 if (created == null) {
-                    created = new AndroidCaseXmlParser(parser, tallies, true, sandbox.getCaseStorage(), generator);
+                    created = new AndroidCaseXmlParser(parser, tallies, true, sandbox.getCaseStorage(), generator) {
+
+                        @Override
+                        public void onIndexDisrupted(String caseId) {
```
Comment (phillipm) [inline app/src/org/commcare/xml/AndroidTransactionParserFactory.java]: Tests complaining that this override isn't valid

Comment (ctsims) [inline app/src/org/commcare/xml/AndroidTransactionParserFactory.java]: ordering issue due to 
https://github.com/dimagi/commcare/pull/265
dependency

Comment (phillipm) [inline app/src/org/commcare/xml/AndroidTransactionParserFactory.java]: if you put
```
cross-request: https://github.com/dimagi/commcare/pull/265
```
in the description of the PR then jenkins will run the tests with the correct `commcare` branch

Comment (phillipm) [inline app/src/org/commcare/xml/AndroidTransactionParserFactory.java]: The regex that detects cross requests is brittle, only recognizing `cross-request: link`. Feel free to extend it: https://github.com/dimagi/mobile-deploy/blob/master/checkout_cross_request_repo.py#L22

**Comment by phillipm** [comment]
@ctsims Can you add documentation for this feature flag in the '[CommCare Android Developer Options](https://confluence.dimagi.com/display/ccinternal/CommCare+Android+Developer+Options)' confluence doc?

**Comment by ctsims** [comment]
Documented here: https://confluence.dimagi.com/display/ccinternal/CommCare+Android+Developer+Options?src=search

**Pull request merged**